rj_gameplay/rj_gameplay/play/line_up.py

```diff
@@ -1,75 +1,47 @@
-        self.right_x = -1.5
-        self.start_y = 2.0
-        self.y_inc = 0.3
-        self.move_right = move_tactic.Move(np.array([self.right_x, self.start_y]))
-        self.move_left = move_tactic.Move(np.array([self.left_x, self.start_y]))
-        self.role_assigner = NaiveRoleAssignment()
+        super().__init__()
 
-    def compute_props(self, prev_props):
-        pass
+        # super simple FSM
+        # TODO: use FSM class (or at least don't use string literals)
+        self.state = "init"
 
     def tick(
         self,
-        world_state: rc.WorldState,
-        prev_results: role.assignment.FlatRoleResults,
-        props,
-    ) -> Tuple[
-        Dict[Type[tactic.SkillEntry], List[role.RoleRequest]],
-        List[tactic.SkillEntry],
-    ]:
-        # Get role requests from all tactics and put them into a dictionary
-        role_requests: play.RoleRequests = {}
-        if self.move_right.is_done(world_state):
-            role_requests[self.move_left] = self.move_left.get_requests(
-                world_state, None
-            )
-        else:
-            role_requests[self.move_right] = self.move_right.get_requests(
-                world_state, None
-            )
-        # Flatten requests and use role assigner on them
-        flat_requests = play.flatten_requests(role_requests)
-        flat_results = self.role_assigner.assign_roles(
-            flat_requests, world_state, prev_results
-        )
-        role_results = play.unflatten_results(flat_results)
-
-        # Get list of all skills with assigned roles from tactics
-        skill_dict = {}
-        if self.move_right.is_done(world_state):
-            skills = self.move_left.tick(world_state, role_results[self.move_left])
-            skill_dict.update(role_results[self.move_left])
-        else:
-            skills = self.move_right.tick(world_state, role_results[self.move_right])
-            skill_dict.update(role_results[self.move_right])
-        # skills = self.move_right.tick(role_results[self.move_right]) + self.move_left.tick(role_results[self.move_left])
-        # skill_dict = {}
-        # skill_dict.update(role_results[self.move_right])
-        # skill_dict.update(role_results[self.move_left])
-
-        return (skill_dict, skills)
-
-    def is_done(self, world_state):
-        return self.move_left.is_done(world_state)
+        world_state: stp.rc.WorldState,
+    ) -> List[RobotIntent]:
+
+        if self.state == "init":
+            self.prioritized_tactics.append(line_tactic.LineTactic(world_state))
+            self.assign_roles(world_state)
+            self.state = "line_up"
+            return self.get_robot_intents(world_state)
+        elif self.state == "line_up":
+            if self.prioritized_tactics[0].is_done(world_state):
+                self.state = "done"
+            return self.get_robot_intents(world_state)
+        elif self.state == "done":
+            # TODO: does this state need to exist?
```

Comment:
Ongoing debate whether one situation = one play or if situations should have the ability to toggle plays. I'm of the 1:1 camp which is why I left this comment (since if 1:1 the situation will end before the play does, thereby switching the play)